### ISSUE TITLE
Deprecated dependencies end up in user projects

### ISSUE THREAD

**Issue by Artur-** [GoogleChrome/workbox#3228]
**Library Affected**:
workbox-build@7

**Browser & Platform**:
All

**Issue or Feature Request Description**:
Installing workbox-build@7 results the deprecation warnings being printed:
```
npm WARN deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
```
The first one would be solved by #3170

The second one has two causes, as shown by `npm ls --all`:
```
  ├─┬ @rollup/plugin-replace@2.4.2
  │ ├─┬ magic-string@0.25.9
  │ │ └── sourcemap-codec@1.4.8
````

```
  ├─┬ @surma/rollup-plugin-off-main-thread@2.2.3
  │ ├── magic-string@0.25.9 deduped
```

**Comment by pwbriggs** [comment]
Fixed in #3170.

**Comment by tomayac** [comment]
Hi there,

Workbox is [moving to a new engineering team within Google](https://github.com/GoogleChrome/workbox?tab=readme-ov-file#maintenance-update:~:text=Maintenance%20update,owners%20of%20Workbox.). As part of this move, we're declaring a partial bug bankruptcy to allow the new team to start fresh. We realize this isn't optimal, but realistically, this is the only way we see it working. For transparency, here're the criteria we applied:

- For PRs, we closed [everything that doesn't target the current `v7` branch](https://github.com/GoogleChrome/workbox/pulls?q=is%3Apr+is%3Aclosed+-base%3Av7).
- For Issues, we closed [everything that was created before 2024](https://github.com/GoogleChrome/workbox/issues?q=is%3Aissue+is%3Aclosed+created%3A%3C2024-01-01+). 

Thanks, and we hope for your understanding!
The Workbox team
